aptos-move/framework/aptos-stdlib/sources/cryptography/bulletproofs.move

```diff
@@ -0,0 +1,225 @@
+
+    /// The input value given to the prover is too large.
+    const E_VALUE_OUTSIDE_RANGE: u64 = 2;
+
+    /// The range proof system only supports proving ranges of type [0, 2^{bits}) where bits \in {8, 16, 32, 64}.
+    const E_RANGE_NOT_SUPPORTED: u64 = 3;
+
+    //
+    // Structs
+    //
+
+    /// Represents a zero-knowledge range proof that a value committed inside a Pedersen commitment lies in [0, 2^{MAX_RANGE_BITS}).
+    struct RangeProof has copy, drop, store {
+        bytes: vector<u8>
+    }
+
+    //
+    // Public functions
+    //
+
+    /// Returns the maximum # of bits the range proof system can verify proofs for.
+    public fun get_max_range_bits(): u64 {
+        MAX_RANGE_BITS
+    }
+
+    /// Deserializes a range proof from a sequence of bytes.
+    public fun range_proof_from_bytes(bytes: vector<u8>): RangeProof {
+        // TODO: Can we do any pre-validation here?
+        RangeProof {
+            bytes
+        }
+    }
+
+    /// Returns the byte-representation of a range proof.
+    public fun range_proof_to_bytes(proof: &RangeProof): vector<u8> {
+        proof.bytes
+    }
+
+    #[test_only]
+    /// Computes a range proof for the Pedersen commitment to 'val' with 'randomness', under the default Bulletproofs
+    /// commitment key; see `pedersen::new_commitment_for_bulletproof`. Returns the said commitment too.
+    ///  Only works for `num_bits` \in {8, 16, 32, 64}.
+    public fun prove_range(val: &Scalar, r: &Scalar, num_bits: u64, dst: vector<u8>): (RangeProof, pedersen::Commitment) {
+        let (bytes, compressed_comm) = prove_range_internal(scalar_to_bytes(val), scalar_to_bytes(r), num_bits, dst);
+        let point = ristretto255::new_compressed_point_from_bytes(compressed_comm);
+        let point = &std::option::extract(&mut point);
+
+        (
+            RangeProof { bytes },
+            pedersen::new_commitment_from_compressed(point)
+        )
+    }
+
+    /// Verifies a zero-knowledge range proof that the value `v` committed in `com` (under the default Bulletproofs
+    /// commitment; see `pedersen::new_commitment_for_bulletproof`) satisfies $v \in [0, 2^{num_bits})$. Only works
+    /// for `num_bits` \in {8, 16, 32, 64}.
+    public fun verify_range_proof(com: &pedersen::Commitment, proof: &RangeProof, num_bits: u64, dst: vector<u8>): bool {
+        verify_range_proof_internal(point_to_bytes(&pedersen::commitment_as_compressed_point(com)), proof.bytes, num_bits, dst)
+    }
+
+    /// Verifies a zero-knowledge range proof that the value `v` committed in `com` (as v * val_base + r * rand_base,
+    /// for some randomness `r`) satisfies $v \in [0, 2^{num_bits})$. Only works for `num_bits` \in {8, 16, 32, 64}.
+    public fun verify_range_proof_custom_ck(
+        com: &pedersen::Commitment,
+        val_base: &RistrettoPoint, rand_base: &RistrettoPoint,
+        proof: &RangeProof, num_bits: u64, dst: vector<u8>): bool
+    {
+        verify_range_proof_custom_ck_internal(
```

Comment:
You should feature guard this.

Reply:
Will do. I've commented on the guidelines for how to do this, since it's no longer clear how to test these in Move once they are feature-guarded.